modules/core/src/buffer_area.cpp

```diff
@@ -19,7 +19,7 @@ class BufferArea::Block
 private:
     inline size_t reserve_count() const
     {
-        return alignment / type_size - 1;
+        return (size_t)alignment / type_size - 1U;
```

Comment:
the cast operator will bind tighter than division.  Why not:
`(size_t)(alignment / type_size - 1U);`